docs/source/links/state.md

```diff
@@ -0,0 +1,295 @@
+---
+title: apollo-link-state
+description: Manage your local data with Apollo Client
+---
+
+Managing remote data from an external API is simple with Apollo Client, but
+where do we put all of our data that doesn't fit in that category? Nearly all
+apps need to centralize client-side data from user interactions and device APIs
+somewhere!
+
+In the past, Apollo users stored their application's local data in a separate
+Redux or MobX store. With `apollo-link-state`, you no longer have to maintain a
+second store for local state. Your Apollo Client cache is your single source of
+truth that holds all of your local data alongside your remote data. To access or
+update your local state, you use GraphQL queries & mutations just like you would
+for data from a server.
+
+When you use Apollo Client to manage your local state, you get all of the same
+benefits you know and love like caching, optimistic UI, and persistence without
+having to set these features up yourself. 🎉 On top of that, you also benefit
+from the excellent developer experience of Apollo DevTools for painless
+debugging & full visibility into your store.
+
+<h2 id="start">Quick start</h2>
+
+To get started, install `apollo-link-state` from npm:
+
+```bash
+npm install apollo-link-state --save
+```
+
+The rest of the instructions assume that you already [set up Apollo
+Client](/docs/react/basics/setup.html#installation) in your application. After
+you install the package, you can create your state link by calling
+`withClientState` and passing in a resolver map. A resolver map describes how to
+retrieve and update your local data.
+
+Let's look at an example where we're using a GraphQL mutation to update whether
+our network is connected with a boolean flag:
+
+```js
+import { withClientState } from 'apollo-link-state';
+
+const stateLink = withClientState({
+  Mutation: {
+    networkStatus: (_, { isConnected }, { cache }) => {
+      const data = {
+        networkStatus: { isConnected, __typename: 'NetworkStatus' },
+      };
+      cache.writeData({ data });
+      return data;
+    },
+  },
+});
+```
+
+To hook up your state link to Apollo Client, concatenate it to the other links
```

Comment:
I'd really love to see if we could recommend a single way to combine links, like `from`?

```
import { from } from `apollo-link`;

const link = from([
  stateLink,
  httpLink,
]);
```

Looks a lot clearer to me, but maybe it's just me?

Reply:
I'm down to switch it to `from`, I agree that we need to be more consistent. Do you think `from` is more intuitive than `ApolloLink.from`? I've always thought importing `concat` & `from` from the `apollo-link` package is weird since concat & from are array methods in JS, but maybe that's just me.

Reply:
I don't care much which one we recommend, but I'd love to recommend exactly one and use it in all examples! I find the static method much more intuitive than `link.concat(link)` but not opinionated about whether it should be attached to `ApolloLink`.

Reply:
Fixed 👍